cdap-common/src/main/java/co/cask/cdap/common/conf/Constants.java

```diff
@@ -611,6 +611,8 @@
       ImmutableMap.of(Constants.Metrics.Tag.NAMESPACE, NamespaceId.SYSTEM.getNamespace(),
                       Constants.Metrics.Tag.COMPONENT, Constants.Service.TRANSACTION);
 
+    public static final String EMIT_PROGRAM_CONTAINER_METRICS_PREFERENCE = "system.metrics.enabled";
```

Comment:
Define this in `SystemArguments` instead and add a method in there `boolean isMetricsEnabled(Map<String, String> args)`.